WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift

```diff
@@ -48,6 +48,27 @@ class CommentDetailViewController: UITableViewController {
         return cell
     }()
 
+    private lazy var commentService: CommentService = {
+        return .init()
+    }()
```

Comment:
`CommentService` should have an `NSManagedObjectContext`. Since `CommentService` is now used multiple times, maybe this init can use the MOC so it doesn't have to be done elsewhere?


```
private lazy var commentService: CommentService = {
    CommentService(managedObjectContext: ContextManager.shared.mainContext)
}()
```

And then these lines can be removed from `updateComment()`:
```
let context = ContextManager.sharedInstance().mainContext
let commentService = CommentService(managedObjectContext: context)
```

Reply:
Whoops, good catch there! I just assumed that creating the service with an empty init would _just_ work. Apparently, it did, because the method that I'm using didn't touch the MOC at all 😅 . But it'll cause trouble when updating the comment since the method did access the `managedObjectContext` property.

This is fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17246/commits/9950653e19a2e03780646076dd521cd064718b98!